### ISSUE TITLE
declaration-bang-space-before not fired on !default, !optional and !global

### ISSUE THREAD

**Issue by alan-agius4** [stylelint/stylelint#939]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

e.g. "A bug where declaration-bang-space-before is not being fired"

> Which rule, if any, is this issue related to?

e.g. `declaration-bang-space-before`: always

> What CSS is needed to reproduce this issue?

e.g.

```css
$test: #fff!default;
$test: #fff!optional;
$test: #fff!global;
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
    "declaration-bang-space-before": "always",
  }
}
```

> Which version of stylelint are you using?

e.g. `4.4.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

e.g. "PostCSS"

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

e.g. "SCSS"

> What did you expect to happen?

e.g. "Warning fired"

> What actually happened (e.g. what warnings or errors you are getting)?
No Warnings
